src/vuex/modules/connection.js

```diff
@@ -20,26 +18,23 @@ export default function() {
   }
 
   const actions = {
-    async checkForPersistedNetwork({ dispatch, commit }) {
+    async checkForPersistedNetwork({ dispatch, commit }, apollo) {
       const network = JSON.parse(localStorage.getItem(`network`))
-      dispatch(`checkAvailableNetworks`).then( async result => {
-        let availNetworks = result.data.data.networks
-        let availNetworksID = Object.values(availNetworks).map(
-          network => network.id
-        )
-        // Little hack to reorder the array so 'cosmos-hub-mainnet' stays at the beginning
-        if (availNetworks.length > 1) {
-          availNetworks = availNetworks.sort( (n1, n2) => (n1.testnet > n2.testnet) ? 1 : -1)
-        }
-        if (network && availNetworksID.includes(network)) {
-          await commit(`setNetworkId`, network)
-        } else if (network && !availNetworksID.includes(network)) {
-          await dispatch(`setNetwork`, availNetworks[0])
-        }      
-      }).catch( error => {
-        console.error('Error', error)
+      const { data } = await apollo.query({
+        query: Networks
       })
-
+      let availNetworks = Object.values(data.networks).map( network => network.id)
+      if (network && availNetworks.includes(network)) {
+        await commit(`setNetworkId`, network)
+      } else if (network && !availNetworks.includes(network)) {
+        if (data.networks.length > 1) {
```

Comment:
maybe use a safer way by doing a `network.find` for `cosmos-hub-mainet` and if that is not there choose `local-cosmos-hub-testnet`.

Reply:
Sounds cool, yes